code/modules/mob/living/carbon/xenomorph/abilities/shrieker/shrieker_powers.dm

```diff
@@ -0,0 +1,219 @@
+/datum/action/xeno_action/activable/shriek/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/xeno = owner
+	if (!istype(xeno))
+		return
+
+	if (!action_cooldown_check())
+		return
+	if (!check_plasma_owner())
+		return
+
+	if(!xeno.check_state())
+		return
+
+	if (xeno.mutation_type == SHRIEKER_NORMAL)
+		var/datum/behavior_delegate/shrieker/BD = xeno.behavior_delegate
+		if (!istype(BD))
+			return
+		if (!BD.use_internal_acid_ability(shriek_cost))
+			return
+
+	if (curr_effect_type == SHRIEKER_SHRIEK_BUFF)
+
+
+		playsound(xeno.loc, shriek_sound_effectt, 55, 0, status = 0)
+		xeno.visible_message(SPAN_XENOHIGHDANGER("[xeno] emits a piercing shriek!"))
+		xeno.create_shriekwave(color = "#07f707")
+		var/shriek_duration = 150
+		var/image/buff_overlay = get_busy_icon(ACTION_GREEN_POWER_UP)
+		var/mob/living/carbon/Xenomorph/Praetorian/P = owner
+		if (!(P.shriek_status_flags & SHRIEKER_SHRIEK_ENHANCED))
+			P.armor_modifier += XENO_ARMOR_MOD_MED
+			P.damage_modifier += XENO_DAMAGE_MOD_SMALL
+			P.recalculate_armor()
+			P.recalculate_damage()
+			P.shriek_status_flags |= SHRIEKER_SHRIEK_ENHANCED
+			to_chat(src, SPAN_XENOWARNING("Your shriek empowers you to strike harder!"))
+			buff_overlay.flick_overlay(P, 150)
+
+			spawn (shriek_duration)
```

Comment:
turn this into

addtimer(src, CALLBACK(.proc/remove_buff, shriek_duration) etc etc)

look it up for examples